docs/pages/technology/vc-concepts.rst

```diff
@@ -0,0 +1,103 @@
+
+See :ref:`preconfigured concepts <default-concepts>` for a list of supported concepts.
+
+templates
+*********
+
+Credentials are mapped on a concept through templates.
+
+An example concept mapping template looks like this:
+
+.. code-block:: json
+
+    {
+        "id": "<<id>>",
+        "issuer": "<<issuer>>",
+        "type": "ExampleCredential@{1_1},{2_1}",
+        "credentialSubject": {
+            "id": "<<subject>>@{2_2}",
+            "company": {
+                "name": "<<company.name>>@{1_2}",
+                "city": "<<company.city>>"
+            }
+        }
+    }
+
+it follows the syntax of a Verifiable Credential. Only the parts of the credential that are mapped or have a fixed value are in the template.
+The example template above introduces the `company` concept. That name must also be used in the APIs.
+A concept is automatically derived from any *conceptValue* with a `.` in it. Any field that is encapsulated between `<<` and `>>` is called a *conceptValue*.
+This *conceptValue* is also used in the APIs and CLI. It can be used as JSON like:
+
+.. code-block:: json
+
+    {
+        "company": {
+            "name": "X",
+            "city": "Y"
+        }
+    }
+
+or as key-value:
+
+.. code-block:: bash
+
+    company.name=X
+    company.city=Y
+
+generic fields
+==============
+
+The following template only defines the generic fields. The generic fields are required for each credential. A `type` is also always required.
+
+.. code-block:: json
+
+    {
+        "id": "<<id>>",
+        "issuer": "<<issuer>>",
+        "type": "SomeCredential",
+        "credentialSubject": {
+            "id": "<<subject>>",
+        }
+    }
+
+fixed values
+============
+
+The example also contains a field that is not encapsulated, the `type` field with `ExampleCredential` as value. For the `type` field this is always the case.
+When searching for a credential, the fixed value will always be added.
+So, if you search for `company.name=X` through the API, the underlying query to the DB will add `type=ExampleCredential` to that query.
+This will happen for all fixed values. This is important to know because it will influence which indexes are required.
+
+indices
+=======
+
+When searching for or manipulating records in a DB, it's always important to have the correct indices.
+It doesn't really matter which DB you're using, it'll always require some sort of index to be able to perform.
+
+In the concept templates, we use the `@{1_1},{2_1}` syntax for this. The full regex is:
+
+.. code-block:: text
+
+    ((<<[a-zA-Z\.]+>>)|([a-zA-Z\.]+))(@({[1-9](_[1-9])?})(,{[1-9](_[1-9])?})*)?
+
+Every entry between `{` and `}` represents an index. If contains multiple numbers that it represents a compound index.
+The second number is the place in the compound index. Every combination of numbers must be unique.
```

Comment:
Should we mention whether ordering (so `2_1, 2_2` vs `2_2, 2_1`) is important?

Reply:
That depends on the DB and if a compound index can be used for a single term search. Might be a bit too deep down the rabbit hole.